WcaOnRails/app/views/competitions/_index_competitions_list.html.erb

```diff
@@ -50,4 +43,38 @@
       });
     });
   </script>
+<% else %>
+  <% if current_user&.can_see_admin_competitions? %>
+    <% if competitions.empty? %>
+      <%= render 'no_competitions_found' %>
+    <% else %>
+      <% past_competitions, upcoming_competitions = competitions.partition(&:is_over?) %>
+      <% recent_competitions, distant_competitions = past_competitions.partition { |c| (Date.today - c.end_date).to_i < 30 } %>
+      <% unless upcoming_competitions.empty? %>
+        <div class="col-md-12" id="upcoming-comps">
+          <%= render 'admin_index_table', competitions: upcoming_competitions, title: "Upcoming competitions" %>
+        </div>
+      <% end %>
+      <div class="col-md-12" id="recent-comps">
+        <%= render 'admin_index_table', competitions: recent_competitions, title: "Recent competitions" %>
+      </div>
+      <div class="col-md-12" id="past-comps">
+        <%= render 'admin_index_table', competitions: distant_competitions, title: "Past competitions" %>
+      </div>
+    <% end %>
+  <% else %>
+    <%= alert :warning do
+      "You don't have access to this page!"
+      end %>
+  <% end %>
 <% end %>
+<script>
+$(".toggle-body").on('click', function(){
+  $(this).closest("table").children("tbody").toggle();
+  if($(this).html() == '(show)') {
+    $(this).html("(hide)");
+  } else {
+    $(this).html("(show)");
+  }
+});
```

Comment:
This would be easier to read =)
```js
$('.toggle-body').on('click', function() {
  $(this).closest("table").children("tbody").toggle();
  var newMessage = $(this).html() == '(show)' ? '(hide)' : '(show)';
  $(this).html(newMessage);
});
```
Is there any reason for not putting this in the appropriate js file?
Also these parentheses looks kinda weird to me, but this could be personal feeling.